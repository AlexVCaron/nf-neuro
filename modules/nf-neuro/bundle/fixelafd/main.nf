process BUNDLE_FIXELAFD {
    tag "$meta.id"
    label 'process_single'

    container "scilus/scilpy:2.2.0_cpu"

    input:
        tuple val(meta), path(bundles), path(fodf)

    output:
        tuple val(meta), path("*_afd_metric.nii.gz"), emit: fixel_afd
        path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    for bundle in $bundles;
        do\
        bname=\$(basename \${bundle} .trk)
        scil_bundle_mean_fixel_afd \$bundle $fodf ${prefix}__\${bname}_afd_metric.nii.gz
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    scil_bundle_mean_fixel_afd -h

    touch ${prefix}_test_afd_metric.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
