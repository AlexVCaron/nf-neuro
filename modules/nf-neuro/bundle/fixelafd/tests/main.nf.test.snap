{
    "bundle - fixelafd": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__tracking_afd_metric.nii.gz:md5,01ac9796dc0ff6861c2992d513d690d3"
                    ]
                ],
                "1": [
                    "versions.yml:md5,219de1df0d294327a5559c161a0f0dff"
                ],
                "fixel_afd": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__tracking_afd_metric.nii.gz:md5,01ac9796dc0ff6861c2992d513d690d3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,219de1df0d294327a5559c161a0f0dff"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-17T17:38:11.324780769"
    },
    "bundle - fixelafd - stub-run": {
        "content": [
            [
                "versions.yml:md5,219de1df0d294327a5559c161a0f0dff"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-17T17:38:22.322152638"
    }
}